From: Christian Marillat <marillat@debian.org>
Date: Tue, 15 May 2012 04:48:54 -0300
Subject: Remove embedded/downloaded copies of various libraries
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Once this is acceptable, this should be forwarded upstream so that the
use of system libraries is an option, instead of downloading libraries
and programs at build time.

Forwarded: no
Last-Update: 2012-05-15
Reviewed-by: Rogério Theodoro de Brito <rbrito@ime.usp.br>
Signed-off-by: Rogério Theodoro de Brito <rbrito@ime.usp.br>
---
 gtk/src/Makefile.am    |    4 ++--
 make/include/main.defs |   26 +++++---------------------
 test/module.defs       |   10 +++-------
 3 files changed, 10 insertions(+), 30 deletions(-)

diff --git a/gtk/src/Makefile.am b/gtk/src/Makefile.am
index 9c1dbd2..8a0be50 100644
--- a/gtk/src/Makefile.am
+++ b/gtk/src/Makefile.am
@@ -10,7 +10,7 @@ else
 HB_LIBS= \
 	-lhb -la52 -lmkv -lavformat -lavcodec -lavutil -ldca -ldvdnav -ldvdread \
 	-lfaac -lmp3lame -lmpeg2 -lvorbis -lvorbisenc -logg -lsamplerate \
-	-lx264 -lmp4v2 -lswscale -ltheora -lz \
+	-lx264 -lmp4v2 -lswscale -ltheoraenc -ltheoradec -lz -lxml2 \
 	-lbz2 -lpthread -lbluray -lass -lfontconfig -lfreetype
 endif
 
@@ -108,7 +108,7 @@ ghb_LDFLAGS = \
 	-mwindows -Wl,--export-dynamic -Wl,--exclude-libs,ALL
 else
 ghb_LDFLAGS = \
-	-Wl,--export-dynamic -Wl,--exclude-libs,ALL
+	-Wl,--export-dynamic -Wl,--exclude-libs,ALL -Wl,-z,defs -Wl,--as-needed
 endif
 
 ghb_LDADD = $(HB_LIBS) $(GHB_LIBS)
diff --git a/make/include/main.defs b/make/include/main.defs
index e68f679..9770de1 100644
--- a/make/include/main.defs
+++ b/make/include/main.defs
@@ -8,8 +8,6 @@ include $(SRC/)make/include/tool.defs
 
 ###############################################################################
 
-MODULES += contrib/a52dec
-
 ifneq (,$(filter $(BUILD.system),cygwin mingw))
 ifneq ($(HAS.bz2),1)
     MODULES += contrib/bzip2
@@ -20,25 +18,8 @@ ifneq (,$(filter $(BUILD.system),darwin cygwin mingw))
     MODULES += contrib/fribidi
 endif
 
-
-MODULES += contrib/faac
-MODULES += contrib/ffmpeg
-MODULES += contrib/fontconfig
-MODULES += contrib/freetype
-MODULES += contrib/lame
-MODULES += contrib/libass
-MODULES += contrib/libdca
 MODULES += contrib/libdvdread
 MODULES += contrib/libdvdnav
-MODULES += contrib/libbluray
-MODULES += contrib/libmkv
-MODULES += contrib/libogg
-MODULES += contrib/libsamplerate
-MODULES += contrib/libtheora
-MODULES += contrib/libvorbis
-MODULES += contrib/libxml2
-MODULES += contrib/mp4v2
-MODULES += contrib/mpeg2dec
 
 ifneq (,$(filter $(BUILD.system),mingw))
 ifneq ($(HAS.pthread),1)
@@ -46,8 +27,6 @@ ifneq ($(HAS.pthread),1)
 endif
 endif
 
-MODULES += contrib/x264
-
 ifneq (,$(filter $(BUILD.system),cygwin mingw))
 ifneq ($(HAS.iconv),1)
     MODULES += contrib/libiconv
@@ -83,6 +62,11 @@ ifeq (1-linux,$(FEATURE.gtk)-$(BUILD.system))
     MODULES += gtk
 endif
 
+ifeq (1-kfreebsd,$(FEATURE.gtk)-$(BUILD.system))
+    ## build gtk when gtk+linux
+    MODULES += gtk
+endif
+
 ifeq (1,$(FEATURE.local_yasm))
     MODULES += contrib/yasm
 endif
diff --git a/test/module.defs b/test/module.defs
index 5030e6f..69f9166 100644
--- a/test/module.defs
+++ b/test/module.defs
@@ -10,10 +10,11 @@ TEST.c.o = $(patsubst $(SRC/)%.c,$(BUILD/)%.o,$(TEST.c))
 TEST.exe = $(BUILD/)$(call TARGET.exe,$(HB.name)CLI)
 
 TEST.libs = $(LIBHB.a) $(foreach n, \
-        a52 ass avcodec avformat avutil dca dvdnav dvdread faac fontconfig freetype mkv mpeg2 mp3lame mp4v2 \
-        ogg samplerate swscale theora vorbis vorbisenc x264 xml2 bluray, \
+        dvdnav dvdread, \
         $(CONTRIB.build/)lib/lib$(n).a )
 
+TEST.libs += -lass -lfaac -lmp3lame -lsamplerate -lvorbisenc -ltheoraenc -ltheoradec -lvorbis -lx264 -lmkv -lz -lbz2 -lpthread -ldl -lmp4v2 -lxml2 -ldca -la52 -lbluray -lavcodec -lavformat -lavutil -lswscale -lmpeg2
+
 TEST.install.exe = $(DESTDIR)$(PREFIX/)bin/$(notdir $(TEST.exe))
 
 ifeq (1,$(PTHREADW32.enabled))
@@ -32,11 +33,6 @@ ifeq (1,$(ZLIB.enabled))
 else
 	TEST.GCC.l += z
 endif
-ifeq (1,$(FRIBIDI.enabled))
-    TEST.libs += $(CONTRIB.build/)lib/libfribidi.a
-else
-	TEST.GCC.l += fribidi
-endif
 
 ###############################################################################
 
